### ISSUE TITLE
Whatevergreen / GPU crash and screen goes black 

### ISSUE THREAD

**Issue by MSTRKKRFT** [acidanthera/bugtracker#2021]
Hi, 

I've a problem on my hack since a few weeks : it crashes, both screens goes black (but hack is still on), and then restarts randomly after some time, under low/no load most of the time.  My hack was running fine without any crash/issue for more than a year, and this problem happened suddenly. 

And after a few days of testing the GPU, updating/changing vBIOS and other things, I tried to **use an older version of Whatevergreen** from a period of time I knew my hack was running fine. **So I replaced version 1.5.8 with version 1.4.6 and since that everything seems to work without issue.** I had maximum 6/8h of uptime before the GPU would crash with 1.5.8 and at the time I'm writing here the hack is running fine for about 24h with 1.4.6. I will keep it on as long as possible to see the evolution and if the problem is really solved or not. 

I don't have the time these days to find the exact version of WEG from which the problem did start to occur but I could provide logs or anything else if needed. I don't know much about debug/programming and the "under the hood" part of OpenCore, but I did made a few hacks with OpenCore right now and I think I know most about it for desktop hacks at least. 

Here are my specs : 
- MacOS Version : BigSur 11.6.6 (20G624)
- Hardware : i7-9700k (no overclock) + 64Gb RAM (4*16) @3200Mhz + Gigabyte z390-AORUS-ELITE + Vega64 8Go + Fenvi T919 + Samsung 970 EVO 500Gb
- OC + OpenCanopy : 8.0.0 
- Kexts (last update, except for WEG of course) : VirtualSMC + Lilu + WhateverGreen + AppleALC + IntelMausiEthernet + NVMeFix
- SSDT (created with SSDTTime) : EC + PLUG + PMC + USBX 
- USBMap has been done with Hackintool (using USBPorts.kext) 
- My config.plist : https://anonfiles.com/Z2Yao8i0y3/config_plist

I've never used OpenCore/Lilu debug, but if needed I could make a usb stick with these to provide more informations about the issue. 

Thanks for your help, 
MSTRKKRFT

**Comment by MSTRKKRFT** [comment]
Hi, thanks for your reply @vit9696
I'm really sorry but I don't know what you mean by "the exact commit". Sorry for my noobiness, as I said I don't know much about dev/debug, but I really want to help.

**Comment by telepati** [comment]
@MSTRKKRFT download from here;
https://dortania.github.io/builds/?product=WhateverGreen&viewall=true

**Comment by MSTRKKRFT** [comment]
Never mind, the problem is back after a few days... I've switched back to last WEG version. 

Still really strange, I hope the GPU is not faulty but It's unlikely as it's working great under windows 11 when I'm gaming. I'm doing a lot of video editing/animation on Premiere/After Effects on this computer, and what is strange is that KP occurs almost only when the computer is idle, not when I'm working on it with heavy CPU/GPU load. 

I've investigated a bit more on the issue and it seems to be a common MacOS problem (not related to OpenCore/hackintosh). I've found topics on apple forums about that KP "no successful checkins from com.apple.WindowServer in 120 seconds" 

Here is the full log, I've tried all solutions I've found on the web (disabling SMB/sharing options, disabling all energy saver options except from "prevent computer from sleeping automatically", also using KeepingYouAwake all the time, removing Terminal from Dev Tools in Privacy settings) with no luck. 
[log-kp.txt](https://github.com/acidanthera/bugtracker/files/8785878/log-kp.txt)

I know this is definitely not a place to ask for help about fixing MacOS problems but I just don't know where to go now (and if I post on apple forum saying I'm using a hackintosh, well...), as apple devs seems to bury one's head in the sand about this issue... And sometimes I honestly think that opencore/hackintosh devs seems to know more about apple systems than apple itself... 

So here is my bottle in the sea! If it's too off-topic I will close the issue and you will never heard about me again here, but if you just could give me a glimmer of hope on this it would already be awesome. 

Maybe updating to Monterey could fix the issue ? I've never had any problem when I was using Mojave but it's too old to keep it as a daily driver now.
